packages/stores/types.ts

```diff
@@ -251,6 +251,7 @@ type TCommonStoreError = {
 
 type TCommonStore = {
     error: TCommonStoreError;
+    services_error: { code: string; message: string; type: string } | Record<string, never>;
```

Comment:
@farzin-deriv I don't understand it completely, we are adding `Record<string, never>` just for ability to write empty object in `mockStore.ts`? I thought mockStore must satisfy types we described in `stores/types.ts`, not vice versa. Based on this types `services_error` can be empty object, but it's not true. If you look at all objects in `mockStore.ts` you can see, for example, object `error`:

error: {
                app_routing_history: [],
                header: '',
                message: '',
                type: '',
                redirect_label: '',
                redirect_to: '',
                should_clear_error_on_click: false,
                should_show_refresh: false,
                redirectOnClick: jest.fn(),
                setError: jest.fn(),
            },

Please explain me why we can't write in `mockStore.ts` like this and delete type `Record<string, never>` if we did like this way all the time:
`services_error: {.code: '', message: '', type: ''}`

Reply:
@sergei-deriv Hey buddy, Imagine our stores were not in the `@deriv/core` and been somewhere shared like `@deriv/stores`, So you could import `CommonStore` in your test file.

Then you wouldn't need mocking things at all, In each of your test cases you would do `new CommonStore()` to instantiate a fresh `CommonStore` for that test case (which could be put in `beforeAll`).

So after `new CommonStore()`, What would be the value for `services_error`? The default value for it is `{}`, Right? So without doing any action on `CommonStore` this value is an empty object. I do agree the value is wrong and maybe shoud've been undefined by default, But changing it now might create lot's of side effects.

Now let's come back to reality from the imagination 😁 Our stores are in `@deriv/core` and not accessible and testable, The only way we can test things now is through mocking things. The `mockStore` function isn't doing anything to be honset, It's just a huge object of the default values in the stores so we don't have to repeat them everywhere.

So to simulate `new CommonStore()`, We should put the same initial values in `mockStore` so in each test case when we create a new mock it's like we are creating a new store.

I'm not sure if I was able to explain it clearly, Let's have a call and discuss this if needed 🙏🏻

Reply:
@farzin-deriv Hi. I have some thoughts about your answer :)

You said: **So after new CommonStore(), What would be the value for services_error? The default value for it is {}, Right?**

In my worldview - no, not right! It could be or `{ code: string, message: string, type: string }` OR `undefined`, why it should be an empty object? If we don't have an error we just don't receive an error in server response, we don't receive an empty object. I understand we have an empty object in store as initial value and you afraid to create a side effects, but, in my opinion, I'm sure on 100%, we have to fix this type issue instead of creating type for empty object.

About mockStore, I understand it just returns an object with initial values and combine it with object passed as argument.

You said: **So to simulate new CommonStore(), We should put the same initial values in mockStore so in each test case when we create a new mock it's like we are creating a new store.**

I understand what you mean and agree with you about creating a new object in every test case, but I don't see any problem to do this in `beforeEach`, in my opinion, it's the same behaviour and allow to avoid code duplication.

Anyway, thanks for your response and your time